pslab/serial_handler.py

```diff
@@ -265,6 +268,30 @@
         self._write_log(version, "RX")
         return version.decode("utf-8")
 
+    def get_firmware_version(self) -> tuple[int, int, int]:
```

Comment:
Total nonsense. 

> The get_firmware_version method adds unnecessary complexity and redundancy to the code. It duplicates the pattern of sending and receiving bytes already established in the class. Consider refactoring to use existing methods for consistency.

It is already using the correct interface: `send_byte` and `get_byte`.

> The error handling for serial.SerialException might not be the best approach for a missing firmware version;

If a firmware version query is put to a firmware version which does not support it, there will be no response. This will manifest as a `serial.SerialException` raised in `_receive`.

> instead, check for support before attempting to read it.

Checking for firmware  support is literally the only purpose of the `get_firmware_version` method added in this PR. There is no way to check for support other than to try the query.

> initializing self.firmware in the constructor increases complexity and assumes immediate availability, which might not always be the case.

Caching the firmware version as an instance variable mirrors the way the hardware version is cached in `self.version`.